File: python_practice/test/bin.pl
#!/bin/perl

use strict;
use warnings;

mkdir "test_filespl";

my $output_file;
my $bin_file;

foreach my $bin_file (<*.bin>)
{
	 $output_file = "bin_filespl/$bin_file.txt";
	`xxd $bin_file > $output_file`;
}

print " success"

File: python_practice/test/bin.py
import os

os.makedirs("test_filepy",exist_ok=True)

for bin_file in os.listdir():
    if bin_file.endswith(".bin") and os.path.isfile(bin_file):
        output_file = f"test_filepy/{bin_file}.txt"
        os.system(f"xxd {bin_file} > {output_file}")

print("success")
